bemani/backend/jubeat/base.py

```diff
@@ -172,10 +190,11 @@ def update_score(
             self.PLAY_MEDAL_EXCELLENT,
         ]:
             raise Exception(f"Invalid medal value {medal}")
-
+        if hard_mode:
+            chart += 3
```

Comment:
Prefer an explicit lookup to adding magic numbers. Makes the code searchable and doesn't bake an impossible-to-find assumption into the current numbering of charts. Also, the base game score saving feature should only be responsible for directly saving/loading a score based on the chart provided. The normal/hard calculation should move to the game and be kept in a game-specific mapping there.

Reply:
This was just the "cleanest" solution I could come up with. My philosophy was to put it here since it lets me implement hard mode saving in each game without worrying about the math there

Reply:
Given hard mode's plumbing through all games, it might actually make sense to move some of it to the base game like you've done here. However, still no magic addition. I'd greatly prefer a mapping, since that's greppable in the code.

Reply:
```python
if hard_mode:
    hard_mode_map = {
        self.CHART_TYPE_BASIC: self.CHART_TYPE_HARD_BASIC,
        self.CHART_TYPE_ADVANCED: self.CHART_TYPE_HARD_ADVANCED,
        self.CHART_TYPE_EXTREME: self.CHART_TYPE_HARD_EXTREME,
    }
    chart = hard_mode_map.get(chart)
```
Do you mean doing something like this to make it more transparent what is going on?